src/path.cpp

```diff
@@ -275,7 +275,7 @@ static void maybe_issue_path_warning(const wcstring &which_dir, const wcstring &
         debug(0, _(L"The error was '%s'."), strerror(saved_errno));
         debug(0, _(L"Please set $%ls to a directory where you have write access."), env_var);
     }
-    write(STDERR_FILENO, "\n", 1);
+    write_ignore(STDERR_FILENO, "\n", 1);
```

Comment:
The previous change to explicitly initialize some size_t vars is fine. I'm less happy with this change. Prior to this there are only four locations that use `write_ignore()`. I'd like input from others but for me I would prefer to modify the callers to explicitly discard the return value. That is, replace `write_ignore(...)` with `(void)write(...)`. While the name of the former function is reasonably clear it adds an unneeded level of indirection and can thus cause confusion.